### DISCUSSION TITLE
Capture freezing on AI-Deck

### DISCUSSION THREAD

**Discussion by dielso** [bitcraze/discussions#256] [Q&A]
Hello!

I've been trying to capture the camera images with the AI-Deck and I've been unsuccessful so far. I've based my code off the [wifi-streamer](https://github.com/bitcraze/aideck-gap8-examples/tree/master/examples/other/wifi-img-streamer) and [test_camera](https://github.com/bitcraze/aideck-gap8-examples/tree/master/examples/other/test_functionalities/test_camera) examples (mostly off the img streamer). When I call a capture (be it async or sync), the function halts infinitely, as if the camera never captures anything:
```
STAB: Ready to fly.
GAP8: Starting camera task...
GAP8: Buffers allocated and camera opened
GAP8: Starting async capture [...]
```
I don't think the problem is in the hardware, since the img streamer works ok.
My capture code is basically structed like this:
```
while(1){
  vTaskDelay(20);

  pi_camera_capture_async(&camera, buff, resolution, pi_task_callback(&task1, capture_done_cb, NULL));
  pi_camera_control(&camera, PI_CAMERA_CMD_START, 0);
  cpxPrintToConsole(LOG_TO_CRTP, "Starting async capture\n");
  xEventGroupWaitBits(evGroup, CAPTURE_DONE_BIT, pdTRUE, pdFALSE, (TickType_t)portMAX_DELAY);
  pi_camera_control(&camera, PI_CAMERA_CMD_STOP, 0);
  cpxPrintToConsole(LOG_TO_CRTP, "Finished async capture\n");
  [...]
}
```
Any help is very much welcome!

**Comment by knmcguire** [comment]
Hi! Since we don't have the full code here are some questions:

- So in anywhere of the code, do you enable the gap8 cluster? 
- Do you send the picture via WiFi?
- If you take away all the other functionalities and have a bare minimum example, what is the problem then

**Reply by dielso** [reply]
Hello!
I'll answer topic-by-topic.
- Bare minimum example:
I removed everything (or almost), except debugging stuff and the capture, and the problem is still the same. Strangely, it did capture one time but when I reflashed it, it didn't, so I'm assuming it was just a bug. I put the code on [this](https://github.com/dielso/aideck-tests) repo.
- Enable the gap8 cluster:
In my last code, I did enable it, but only after the capture for further processing. Why? 
- Do you send the picture via WiFi?
Nope!

Thanks for the answer!

Edit: I removed the camera rotation and it stopped stalling. Soon enough I'll come back to investigate this problem further.

**Reply by knmcguire** [reply]
The reason I asked about the cluster is because of this issue that you might be hitting: https://github.com/bitcraze/aideck-gap8-examples/issues/100. That it works but CPX stalles

Interesting that the camera rotation had something to do with it... where in the code was this placed? Same as the wifi example I suppose?
